vyper/codegen/core.py

```diff
@@ -107,61 +107,72 @@ def _dynarray_make_setter(dst, src):
         store_length = IRnode.from_list(store_length, annotation=ann)
         ret.append(store_length)
 
+        ret.extend(_copy_dynarray_body(dst, src))
+        return ret
+
+    with src.cache_when_complex("darray_src") as (b1, src), get_dyn_array_count(
+        src
+    ).cache_when_complex("darray_count") as (b2, count):
+        ret.append(STORE(dst, count))
+        ret.extend(_copy_dynarray_body(dst, src, loop_count=count))
+        return b1.resolve(b2.resolve(ret))
+
+
+def _copy_dynarray_body(dst, src, loop_count=None):
```

Comment:
is `loop_count` a necessary parameter for this function?

Reply:
If we do not pass in the `loop_count` from `_dynarray_make_setter`, the `test_nested_struct_of_lists` test in `tests/parser/types/test_dynamic_array.py` will pass with the `--no-optimize` flag instead of throwing a compilation error. Should I amend the test?